nautobot/utilities/testing/filters.py

```diff
@@ -67,6 +76,47 @@ def test_invalid_filter(self):
             params = {"ice_cream_flavor": ["chocolate"]}
             self.assertFalse(self.filterset(params, self.queryset).is_valid())
 
+        def test_filters_generic(self):
+            """
+            Tests multiple choice filters declared in `self.generic_filter_tests`. This uses `get_filterset_test_values()` to retrieve
+            a valid set of test data and asserts that the queryset filter matches the filterset's returned queryset. The majority of
+            Nautobot filters use conjoined=False, so the extra logic to support conjoined=True has not been implemented here. TagFilter
+            and similar AND filters are not supported. Multiple tests can be performed for the same filter by adding multiple entries
+            into `generic_filter_tests` with custom field names. For example, to test NaturalKeyOrPKMultipleChoiceFilter, use:
+            generic_filter_tests = (
+                ["filter_name", "field_name__slug"],
+                ["filter_name", "field_name__id"],
+            )
+            """
+            for test in self.generic_filter_tests:
+                filter_name = test[0]
+                field_name = test[-1]
```

Comment:
I know this is a minor nitpick but every time I see indexing, it makes my throat hurt. Since `test` is not actually used, you can just unpack the tuple while iterating:

```suggestion
            for filter_name, field_name in self.generic_filter_tests:
```

Reply:
Nevermind, I see now why you did this. Could you add some extra commenting that these can also be single-member lists thus the negative indexing?

Reply:
I updated the docstrings for the new tests too. Mind taking another look at this file?